### PR TITLE
FOUNDATIONS: Add Student

### LINKED ISSUE #16: FOUNDATIONS: Add Student

### PR DESCRIPTION
**Pull request by ShriHumrudha**
Closes #16

### CODE REVIEW

File: SCMS.Portal.Web/Services/Foundations/Students/StudentService.Exception.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            catch (HttpRequestException httpRequestException)
+            {
+                var failedStudentDependencyException =
+                    new FailedStudentDependencyException(httpRequestException);
+
+                throw CreateAndLogCriticalDependencyException(failedStudentDependencyException);
+            }
+            catch (HttpResponseUrlNotFoundException httpResponseUrlNotFoundException)
+            {
+                FailedStudentDependencyException failedStudentDependencyException =
```
Comment (hassanhabib) [inline SCMS.Portal.Web/Services/Foundations/Students/StudentService.Exception.cs:42]: `var`

File: SCMS.Portal.Web/Services/Foundations/Students/StudentService.Exception.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                    new FailedStudentDependencyException(httpResponseUrlNotFoundException);
+
+                throw CreateAndLogCriticalDependencyException(failedStudentDependencyException);
+            }
+            catch (HttpResponseUnauthorizedException unauthorizedHttpResponseException)
+            {
+                FailedStudentDependencyException failedStudentDependencyException =
```
Comment (hassanhabib) [inline SCMS.Portal.Web/Services/Foundations/Students/StudentService.Exception.cs:49]: `var`

File: SCMS.Portal.Web/Services/Foundations/Students/StudentService.Validation.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            Message = "Value is invalid."
+        };
+
+        private static dynamic IsInvalid(
+            DateTimeOffset firstDate,
+            DateTimeOffset secondDate,
+            string secondParameterName) => new
+            {
+                Condition = firstDate != secondDate,
+                Message = $"Date is not same as {secondParameterName}."
```
Comment (hassanhabib) [inline SCMS.Portal.Web/Services/Foundations/Students/StudentService.Validation.cs:80]: These are logical validations - they live in the API not the portal/client.

Comment (hassanhabib) [reply SCMS.Portal.Web/Services/Foundations/Students/StudentService.Validation.cs:80]: Clients only handle structural validations.

**Review by osp709 — APPROVED** [review]

**Pull request merged**